CHANGELOG.md

```diff
@@ -17,10 +17,11 @@
 ### What's new
 
 - Crates can now use proc-macros without UDL files to export their interface.  See the "Procedural Macros: Attributes and Derives" manual section for details.
-
 - [Custom Types](https://mozilla.github.io/uniffi-rs/proc_macro/index.html#the-unifficustomtype-derive) are now supported for proc-macros, including a very
   low-friction way of exposing types implementing the new-type idiom.
 - Proc-macros: Added support for ByRef arguments
+- Proc-macros: Implemented custom type conversion error handling (https://mozilla.github.io/uniffi-rs/udl/custom_types.html#error-handling-during-conversion)
+- Proc-macros: Error types must now implement `Error + Send + Sync + 'static`, like they do for UDL.
```

Comment:
I don't think this should be a large burden for library consumers, but please speak up if it will affect you.

Reply:
This applies to UDL code though too now.
At least I just pulled in your changes to compile Glean against and noticed that we _don't_ yet implement `std::error::Error`. Easy to solve and we already satisfy the other bounds.

Reply:
We could keep the bound as `Display + Debug + Send + Sync + 'static`, but I thought it was simpler to spell out the requirement as `Error + Send + Sync + 'static` and maybe we'll need the `Error` bound sometime in the future.